### PR TITLE
Add an iptables rule for letting weaveDNS requests in

### LINKED ISSUE #1578: DNS broken on Ubuntu 15.10 desktop (w UFW)
When running Weave on Ubuntu:

```
$ ./weave launch
$ C2=$(./weave run --name c1 -ti ubuntu  /bin/bash)
$ C2=$(./weave run --name c2 -ti ubuntu  /bin/bash)
$ docker attach $C2
# ping c1
ping: unknown host c1
```

The container can `ping` the DNS resolver:

```
# cat /etc/resolv.conf 
nameserver 172.17.42.1
# ping 172.17.42.1
PING 172.17.42.1 (172.17.42.1) 56(84) bytes of data.
64 bytes from 172.17.42.1: icmp_seq=1 ttl=64 time=0.043 ms
64 bytes from 172.17.42.1: icmp_seq=2 ttl=64 time=0.043 ms
64 bytes from 172.17.42.1: icmp_seq=3 ttl=64 time=0.059 ms
64 bytes from 172.17.42.1: icmp_seq=4 ttl=64 time=0.059 ms
...
```

But `ufw` seems to be blocking traffic to the DNS server:

```
$ sudo tcpdump -n -i docker0    "port 53"
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on docker0, link-type EN10MB (Ethernet), capture size 262144 bytes
12:59:08.381052 IP 172.17.0.2.48960 > 172.17.42.1.53: 56666+ A? c1.weave.local. (32)
12:59:13.381821 IP 172.17.0.2.48960 > 172.17.42.1.53: 56666+ A? c1.weave.local. (32)
12:59:18.383369 IP 172.17.0.2.37360 > 172.17.42.1.53: 30668+ A? c1. (20)
12:59:23.388212 IP 172.17.0.2.37360 > 172.17.42.1.53: 30668+ A? c1. (20)
```

It can be seen with

```
$ dmesg 
[ 3029.623661] [UFW BLOCK] IN=docker0 OUT= PHYSIN=vethf203be5 MAC=02:42:b6:97:66:4f:02:42:ac:11:00:02:08:00 SRC=172.17.0.2 DST=172.17.42.1 LEN=60 TOS=0x00 PREC=0x00 TTL=64 ID=52288 DF PROTO=UDP SPT=54743 DPT=53 LEN=40 
```

### PR DESCRIPTION
**Pull request by inercia**
Fixes #1578

### CODE REVIEW

**Comment by rade** [comment]
should be branched off 1.2

**Comment by inercia** [comment]
> why do we need to *insert* the rule rather than appending it?

Because I think we need it before all these rules:

```
Chain INPUT (policy DROP 137 packets, 4384 bytes)
 pkts bytes target     prot opt in     out     source               destination         
 136K   98M ufw-before-logging-input  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
 136K   98M ufw-before-input  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
 7237  424K ufw-after-input  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
 1537 56400 ufw-after-logging-input  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
 1537 56400 ufw-reject-input  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
 1537 56400 ufw-track-input  all  --  *      *       0.0.0.0/0            0.0.0.0/0 
```

**Comment by inercia** [comment]
@rade you are right, we append it. Fixing...

File: weave [hunk 1]
```diff
@@ -1479,6 +1479,9 @@ launch_router() {
         --docker-api "unix:///var/run/docker.sock" "$@")
     with_container_netns_or_die $CONTAINER_NAME setup_router_iface_$BRIDGE_TYPE
     attach_router
+
+    # let DNS traffic to weaveDNS
+    [ -n "$DNS_PORT_MAPPING" ] && add_iptables_rule filter INPUT -i $DOCKER_BRIDGE -p udp --dport 53  -j ACCEPT
```
Comment (rade) [inline weave]: don't we need tcp too?

File: weave [hunk 2]
```diff
@@ -1479,6 +1479,9 @@ launch_router() {
         --docker-api "unix:///var/run/docker.sock" "$@")
     with_container_netns_or_die $CONTAINER_NAME setup_router_iface_$BRIDGE_TYPE
     attach_router
+
+    # let DNS traffic to weaveDNS
```
Comment (rade) [inline weave]: "..., since otherwise it might get blocked by the likes of ufw"

File: weave (hunk 1 above)
Comment (inercia) [inline weave]: Added.

**Comment by inercia** [comment]
It seems I'll have to close this branch as the target branch [cannot be changed](isaacs/github#18) in a PR... :confused:

**Comment by rade** [comment]
yes

**Comment by inercia** [comment]
Tests failing with:

```
>>> Test ./640_proxy_restart_reattaches_test.sh finished after 32.7 secs with error: exit status 1
Proxy restart reattaches networking to containers
d4b5c5a8baf365fca5fc9f68582c4802b8b7d9d115ff42c81b8fbd92feb49d2e
6123910fedad4ebf07e17a5211f33677e461651b929c9eced84675c4813bdc24
c2
weave
sudo: systemctl: command not found
```

:confused:

**Comment by rade** [comment]
> Tests failing

#1588

**Comment by inercia** [comment]
I'll wait until #1588 is fixed, then I'll close this PR and open a new one based on branch `1.2`.

**Comment by rade** [comment]
> I'll wait until #1588 is fixed

Don't. Unless you are going to fix it yourself.

**Pull request closed without merging**